### DISCUSSION TITLE
Multiple ports?

### DISCUSSION THREAD

**Discussion by rmortes** [hollie/tailscale-caddy-proxy#10] [Q&A] [answered]
Hi there! Is there possible to forward multiple ports through a single hostname?

For example, if I had a single service that exposed the ports 80, 90, and 8000, would it be possible to access them using service.tail1234.ts.net, service.tail1234.ts.net:90 and service.tail1234.ts.net:8000?

I'm not quite sure what "Caddy" is, and the reverse proxy documentation doesn't answer my question I think.

Thanks!

**Comment by ksylvan** [comment]
Yes, with the "tailscale funnel" and Magic DNS, you can even expose your containers to the larger network.

I find this approach (from Tailscale Devs) easier than the `tailscale-caddy-proxy` approach:

https://github.com/tailscale-dev/docker-guide-code-examples

**Comment by hollie** [comment, marked as answer]
Hello @rmortes,

if you want to share multiple services on different ports through a single Tailscale hostname then tailscale-caddy-proxy is not what you are looking for. You want to use either the standard Tailscale docker container in that case, or consider adding Tailscale (the binary) directly to the container you are building.

The reason why I created tailscale-caddy-proxy was because:
* I want to be able to share services using different hostnames, without having to remember ports. Using hostnames + the standard HTTPS port gives a better user experience in my opinion.
* at the time I created this container, `tailscale serve` did not yet exist
* I find it much more manageable to define all my tailscale 'virtual hosts' in a single docker compose file running in a separate LXC container on proxmox compared to running 'tailscale serve' on every service I want to share separately.

Best regards,
 Lieven.
